plugins/modules/swim_workflow_manager.py

```diff
@@ -1513,78 +1532,121 @@ def get_diff_tagging(self):
 
-        image_params = dict(
-            image_id=self.have.get("tagging_image_id"),
-            site_id=self.have.get("site_id"),
-            device_family_identifier=self.have.get("device_family_identifier"),
-            device_role=tagging_details.get("device_role", "ALL").upper()
-        )
+        device_roles = ["core", "distribution", "access", "border router", "unknown", "all"]
 
-        response = self.dnac._exec(
-            family="software_image_management_swim",
-            function='get_golden_tag_status_of_an_image',
-            op_modifies=True,
-            params=image_params
-        )
-        self.log("Received API response from 'get_golden_tag_status_of_an_image': {0}".format(str(response)), "DEBUG")
+        for role in device_role.split(','):
+            role = role.strip()
+            device_role_no.append(role)
 
-        response = response.get('response')
-        if response:
-            image_status = response['taggedGolden']
-            if image_status and image_status == tag_image_golden:
+            if role.lower() not in device_roles:
+                self.status = "failed"
+                self.msg = (
+                    "Validation Error: The specified device role '{0}' is not recognized. "
+                    "Please ensure the role matches one of the known device roles: {1}."
+                ).format(role, ", ".join(device_roles))
+                self.log(self.msg, "ERROR")
+                self.result['response'] = self.msg
+                self.check_return_status()
+
+        for role in device_role.split(','):
+            image_params = {
+                "image_id": self.have.get("tagging_image_id"),
+                "site_id": self.have.get("site_id"),
+                "device_family_identifier": self.have.get("device_family_identifier"),
+                "device_role": role.upper()
+            }
+
+            response = self.dnac._exec(
+                family="software_image_management_swim",
+                function="get_golden_tag_status_of_an_image",
+                op_modifies=True,
+                params=image_params
+            )
+            self.log("Received API response from 'get_golden_tag_status_of_an_image': {0}".format(str(response)), "DEBUG")
+
+            api_response = response.get('response')
+            if api_response:
+                image_status = api_response.get('taggedGolden')
+                self.log(image_status)
+                self.log(tag_image_golden)
+
+                if image_status == tag_image_golden:
+                    self.log("inside if")
+                    msg = "SWIM Image '{0}' already tagged as Golden image in Cisco Catalyst Center".format(image_name)
+                    self.log(msg, "INFO")
+                    already_tagged_device_role.append(role)
+
+                if not image_status and image_status == tag_image_golden:
```

Comment:
this should be elif since you have already checked image_status as tag_image_golden in 1584 a

`    elif not image_status and not tag_image_golden:
`

Reply:
updated the code